Add-on to HTMS

educhielle opened the pull request:
<!-- Please review the contributing guidelines before submitting this PR. -->

minj commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/minj/foxtrick/1501)
<!-- Reviewable:end -->

educhielle commented:
sorry, I was having a hard time committing the new files and it was pissing me off
so I decided to create a new pull request

I did some of the modifications you mentioned (the ones I knew how).

1st:
_var luckContent = ( luck > 0 ? "+"+luck+"%" : luck+"%" );_
This was removed. I don't need this anymore because now we have a gauge.

2nd:
I added gauge.js to the lib directory, but I couldn't make the add-on load it from there.

3rd:
_mimimicanvas.style.width = '80px';_
I modified this file _resources/css/htms-statistics.css_, so now the width comes from there

4th:
_var luckText = 'luck'; // replace by something like Foxtrick.L10n.getString('HTMSPrediction.luck');_
now it is var _luckText = Foxtrick.L10n.getString('HTMSPrediction.luck');_
I modified the _/locale/en-GB/foxtrick.properties_ and added _HTMSPrediction.luck=Luck_
But for some reason the "Luck" is showing in lowercase

5th:
_var target = document.getElementById('ft-mimimicanvas');_
replaced by
_var target = doc.getElementById('ft-mimimicanvas');_

minj commented:
Regarding localization as noted in the dev guide:
> Base strings reside in content/foxtrick.properties file. Localized strings are imported from content/locale/*/foxtrick.properties (they are imported from crowdin and should not be touched).
Hence you need to modify `content/foxtrick.properties` rather than `content/locale/en-GB/foxtrick.properties`,

Regarding library loading you could just text-search for another lib like `PluralForm.js`, but there's also a python tool:
```sh
python ./module-update.py add-util -t ext-lib content/lib/gauge.js
```
This would modify the necessary files automatically:
- Info.plist
- content/background.html
- content/bootstrap-fennec.js
- content/bootstrap-firefox.js
- content/preferences.html
- content/scripts-fennec.js
- manifest.json

Of course, if you haven't got python installed, you would do it manually 😉 

There's a bit of a problem in that I forgot to update this script and rethink loading a little since there were some changes made in loading. However, it works fine as is, in this case.

---

A couple of other things I noticed:
-  you used a `htms-mimimi-canvas` class. Normally we start classes and ids with `ft-` as a namespace. As a bonus, all such elements are automatically marked when using "Highlight features".
- gauge license is missing

educhielle commented:
Hello, I have fixed the issues.
The python script worked just fine in Linux. In Windows it generates scripts that have some problems.
In addition, I have found a bug in my add-on and fixed it. The bug was in the simulator, when selecting a previous match of the adversary. It was showing the space for the gauge (but not the gauge) and text "Luck". I have fixed it using:
_var mimimiIsChecked = true && Foxtrick.Pages.Match.hasRatingsTabs(doc);_
It returns true when it is a real match with result, and false when it is a simulation like the pic below. I still need to test in a ongoing match, but I guess it will work. Otherwise, it will be easy to fix.
ps: the true will be replaced by the return from the checkbox in the preferences.
![image](https://cloud.githubusercontent.com/assets/25762736/23414601/f3cd779a-fdf5-11e6-97b1-fac1147369e1.png)

educhielle commented:
Regarding the gauge's license, the owner didn't make it available on his repository.

minj commented:
I guess we need to use the [MIT template](https://en.wikipedia.org/wiki/MIT_License) for Bernard Kobos:

```
Copyright (c) 2017 Bernard Kobos

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
SOFTWARE.
```
I don't have much time during the workdays so I hope to check the code on the weekend

minj commented:
Seems OK so far.

The DevGuide contains a short example how [options are defined and used](minj/foxtrick/blob/master/maintainer/DevGuide.md#an-advanced-module-example).

Please also see the note about [providing defaults](/minj/foxtrick/blob/master/maintainer/DevGuide.md#default-settings)

---

I figure you are using github's built-in upload functionality? It does not make for informative commit messages 😄

Any plans to use a desktop git(hub) client?

Since this is a work in progress I plan to squash everything into a single commit when we are done. You will of course retain the ownership of the final changeset. I hope that's OK with you.

minj commented:
By the way, I am curious what kind of problems you ran into when using the python script on windows.

The obvious thing would be if your git transforms the line endings when checking out the code. I hope you are not using `notepad.exe` for development :smile:

Try (with everything commited)
```
git config --global core.eol lf
git config --global core.autocrlf false
git rm -rf --cached .
git reset --hard HEAD
```

I guess I need to commit a proper `.gitattributes` file

educhielle commented:
I have modified the:
- ./content/matches/htms-prediction.js
- ./content/foxtrick.properties
- ./defaults/preferences/foxtrick.js
and synced using the github desktop. For some reason, it added other files to the commit. It is working fine in chrome, but I still didn't have time to test in the firefox.

Regarding the python script. It ran in Windows, but the files it generated didn't work because of problems with the format of new lines, I guess. I am not using notepad.exe, but I am using Notepad++. :smile:

educhielle commented:
Do you want me to fork again the repository, change only the right files, and submit another pull?

minj commented:
Sorry for sleeping on this for so long.

I was not in the mood and/or luxury to spend time on HT for a while. Let me see if I can solve the remaining issues in a jiffy...

minj commented:
squashed, filtered and rebased your changes as [branch pull/1501](/minj/foxtrick/tree/pull/1501)

educhielle commented:
glad to hear that you are back.. so what is next?

minj commented:
I'll fixup the last problems with the code and merge it to master.

It will reach our beta automatically: https://www.foxtrick.org/download/beta
Then I will upload the strings file for translation in https://crowdin.com/project/foxtrick

I still have some major works to do with Foxtrick back-end before I am satisfied to move everything from beta to stable so that will take time.

You could move on to other issues, if you like. [Searching for issues](https://github.com/minj/foxtrick/issues?q=is%3Aopen+is%3Aissue+label%3Agood-first-issue) labeled `good-first-issue` could you give you some ideas.

educhielle commented:
Sure. I will check the issues to see where I can help.

Should I close this pull request?

minj commented:
I opened #1511 where you can monitor the progress instead
